internal/charmstore/search.go

```diff
@@ -230,6 +236,21 @@ func (si *SearchIndex) update(doc *SearchDoc) error {
 	if err != nil && err != elasticsearch.ErrConflict {
 		return errgo.Mask(err)
 	}
+	if doc.Entity.URL.Series != "" {
+		return nil
+	}
+	// This document represents a multi-series charm it might be
+	// replacing existing documents for previous single-series
+	// charms. Remove any documents that this replaces.
+	for _, series := range doc.Entity.SupportedSeries {
```

Comment:
I think in that case you should still see the precise search result as that is what you would install for precise. There is also the notion of deprecated series which means that eventually those charms would not be added to search in the first place.